### ISSUE TITLE
Move existing front end code under the new front end

### ISSUE THREAD

**Issue by mtakac2** [exercism/exercism.io#2318]
We need to move everything from `frontend` under `client` directory so that we can get rid of lineman and use new pipeline for front end development. Then we can start gradually rewriting existing JS. 
We need to move not only the code itself, but also all JS plugins and libraries (`frontend/vendor`). Most of them should be installed via NPM or bower.

We will also need to move styles. My idea was to create some basic structure with global styles and select styles needed by each module and include it into the module. 
To illustrate this, let's say that we will decide to move our global styles into the `client/app/styles` folder. This folder will include all application specific and application wide styles (colors, layouts, typography, some helpers) and will be loaded first, as a single file. Then `CommentBox` module (`client/app/components/comment_box`), responsible for rendering and events of comment form, will include only styles specific for this module.

- [x] Move all JS under `client` directory and process them using our new pipeline
- [ ] Move all styles under the `client` directory

**Comment by bIgBV** [comment]
This sounds like an interesting approach, regarding the architecture, where can I read more about architecting SPA's ? I have worked with JS before but did not create any major SPA till now.

**Comment by mtakac** [comment]
@bIgBV hmm, that really depends... Do you want to learn about SPAs in general or about building SPAs with Angular? For former just google it - even the wiki article should give you pretty good idea, for later I would suggest to start with the official Angular documentation and tutorials and then just look around - there are lots of great Angular tutorials, screencasts and books out there. (NgNewsletter)[http://www.ng-newsletter.com/] can be quite useful and guys from (Egghead)[https://egghead.io/] are making great screencasts.

BTW From architecture point of view we are talking about (MVVM)[http://en.wikipedia.org/wiki/Model_View_ViewModel] and interesting would also be to take a look at (webpack)[http://webpack.github.io/] which is used to bundle modules together (everything from scripts, to HTML, to styles and assets)

**Comment by kytrinyx** [comment]
I think this is blocked on getting the heroku buildpack working, is that correct?

**Comment by mtakac** [comment]
@kytrinyx Correct.

**Comment by kytrinyx** [comment]
@tejasbubane here's another javascript / lineman related thing.

@mtakac2 I'm so sorry that I failed to figure out how to help get this all sorted out and shipped.

Closing in favor of https://github.com/exercism/exercism.io/issues/3166
